src/sparsezoo/analyze_v2/node_analysis.py

```diff
@@ -38,16 +38,18 @@ def __init__(
         )
 
     def to_dict(self):
-        return NodeAnalysisSchema(
-            name=self.node.name,
-            op_type=self.node.op_type,
-            graph_order=self.graph_order,
-            input=self.node.input,
-            output=self.node.output,
-            params=self.parameter_analysis.to_dict(),
-            ops=self.operation_analysis.to_dict(),
-            mem_access=self.memory_access_analysis.to_dict(),
-        ).dict()
+        if self.parameter_analysis.to_dict() is not None:
+            return NodeAnalysisSchema(
+                name=self.node.name,
+                op_type=self.node.op_type,
+                graph_order=self.graph_order,
+                input=self.node.input,
+                output=self.node.output,
+                params=self.parameter_analysis.to_dict(),
+                ops=self.operation_analysis.to_dict(),
+                mem_access=self.memory_access_analysis.to_dict(),
+            ).dict()
```

Comment:
should be able to use `dict(NodeAnalysisSchema(...))` so you don't need to nest the `to_dict()` calls